modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/BackgroundMBLayer.java

```diff
@@ -179,12 +179,14 @@ public FeatureTypeStyle transformInternal(MBStyle styleContext) {
         rule.setLegendGraphic(new Graphic[0]);
 
         rules.add(rule);
-        return sf.featureTypeStyle(getId(),
+        List<FeatureTypeStyle> style = new ArrayList<>();
```

Comment:
You can use `Collections.singletonList(sf.featureTypeStyle(...))` here instead of creating a new array list. Same for most of the other MBLayer implementations.